src/Access/KerberosInit.cpp

```diff
@@ -0,0 +1,213 @@
+#include <Access/KerberosInit.h>
+#include <Common/Exception.h>
+#include <Common/logger_useful.h>
+#include <Poco/Logger.h>
+#include <Loggers/Loggers.h>
+#include <filesystem>
+#if USE_KRB5
+#include <krb5.h>
+#include <mutex>
+
+using namespace DB;
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int KERBEROS_ERROR;
+}
+}
+
+struct k5_data
+{
+    krb5_context ctx;
+    krb5_ccache in_cc, out_cc;
+    krb5_principal me;
+    char * name;
+    krb5_boolean switch_to_cache;
+};
+
+class KerberosInit
```

Comment:
Although the class is private, I would prohibit making a copy explicitely

Reply:
Added  boost::noncopyable to KerberosInit